renderlib/graphics/RenderGL.cpp

```diff
@@ -100,11 +103,60 @@ RenderGL::renderTo(const CCamera& camera, GLFramebufferObject* fbo)
 
   doClear();
   if (haveScene) {
-    m_image3d->render(camera, m_scene, m_renderSettings);
+    drawSceneObjects(camera);
   }
   fbo->release();
 }
 
+void
+RenderGL::drawSceneObjects(const CCamera& camera)
+{
+  const glm::vec3 volumePhysicalSize = m_scene->m_volume->getPhysicalDimensions();
+  float maxPhysicalDim = std::max(volumePhysicalSize.x, std::max(volumePhysicalSize.y, volumePhysicalSize.z));
+
+  // scene bounds are min=0.0, max=image physical dims scaled to max dim so that max dim is 1.0
+  glm::vec3 sn = m_scene->m_boundingBox.GetMinP();
+  glm::vec3 ext = m_scene->m_boundingBox.GetExtent();
+  CBoundingBox b;
+  b.SetMinP(glm::vec3(ext.x * m_scene->m_roi.GetMinP().x + sn.x,
+                      ext.y * m_scene->m_roi.GetMinP().y + sn.y,
+                      ext.z * m_scene->m_roi.GetMinP().z + sn.z));
+  b.SetMaxP(glm::vec3(ext.x * m_scene->m_roi.GetMaxP().x + sn.x,
+                      ext.y * m_scene->m_roi.GetMaxP().y + sn.y,
+                      ext.z * m_scene->m_roi.GetMaxP().z + sn.z));
+  // LOG_DEBUG << "CLIPPED BOUNDS" << b.ToString();
+  // LOG_DEBUG << "FULL BOUNDS" << m_scene->m_boundingBox.ToString();
+  // draw bounding box on top.
+  // move the box to match where the camera is pointed
+  // transform the box from -1..1 to 0..physicalsize
+  float maxd = (std::max)(ext.x, (std::max)(ext.y, ext.z));
+  glm::vec3 scales(0.5 * ext.x / maxd, 0.5 * ext.y / maxd, 0.5 * ext.z / maxd);
+  // it helps to imagine these transforming the space in reverse order
+  // (first translate by 1.0, and then scale down)
+  glm::mat4 bboxModelMatrix = glm::scale(glm::mat4(1.0f), scales);
+  bboxModelMatrix = glm::translate(bboxModelMatrix, glm::vec3(1.0, 1.0, 1.0));
+  glm::mat4 viewMatrix(1.0);
+  glm::mat4 projMatrix(1.0);
+  camera.getProjMatrix(projMatrix);
+  camera.getViewMatrix(viewMatrix);
+
+  if (m_scene->m_material.m_showBoundingBox) {
+    glEnable(GL_LINE_SMOOTH);
+    glEnable(GL_BLEND);
+    glm::vec4 bboxColor(m_scene->m_material.m_boundingBoxColor[0],
+                        m_scene->m_material.m_boundingBoxColor[1],
+                        m_scene->m_material.m_boundingBoxColor[2],
+                        1.0);
+    m_boundingBoxDrawable->drawLines(projMatrix * viewMatrix * bboxModelMatrix, bboxColor);
+    if (m_scene->m_showScaleBar && camera.m_Projection != ProjectionMode::ORTHOGRAPHIC) {
+      m_boundingBoxDrawable->updateTickMarks(scales, maxPhysicalDim);
+      m_boundingBoxDrawable->drawTickMarks(projMatrix * viewMatrix * bboxModelMatrix, bboxColor);
+    }
+  }
+
+  m_image3d->render(camera, m_scene, m_renderSettings);
```

Comment:
just to separate the "draw stuff" from all the extra framebuffer setup and teardown that wraps it.   Also a bit of DRY since it does appear in two separate functions.